dependencies.yaml

```diff
@@ -130,12 +136,16 @@ dependencies:
       - output_types: [conda, requirements]
         packages:
           - pre-commit
-  doc:
+  docs_build:
     common:
       - output_types: [conda, requirements]
         packages:
-          - pydata-sphinx-theme
+          - doxygen
+          - ipython
           - myst-parser
+          - nbsphinx
+          - numpydoc
+          - pydata-sphinx-theme
```

Comment:
`sphinx` should probably be added explicitly here since we use `sphinx-build` in the scripts.

It's probably being pulled in transitively right now.